src/fsharp/SyntaxTrivia.fsi

```diff
@@ -4,6 +4,34 @@ namespace FSharp.Compiler.SyntaxTrivia
 
 open FSharp.Compiler.Text
 
+[<RequireQualifiedAccess; NoEquality; NoComparison>]
+type ConditionalDirectiveTrivia =
+    | IfDirectiveTrivia of expr:IfDirectiveExpression * range:range
```

Comment:
These names need not include "Trivia" since it is already in the enclosing type name, which is always present due to RQA.  Indeed you could drop "Directive" as well, so
```fsharp
type ConditionalDirectiveTrivia =
    | If of expr:IfDirectiveExpression * range:range
    | Else of range:range
    | EndIf of range:range
```